app/src/routes/editor/EditPanel/inputs/single/ColorInput/index.tsx

```diff
@@ -24,10 +24,14 @@ const ColorInput = observer(
                 return;
             }
             const newValue = elementStyle.getValue(editorEngine.style.selectedStyle?.styles);
-            const hexValue = stringToHex(newValue);
+            const hexValue = Color.from(newValue)?.toHex6() || newValue;
```

Comment:
if `newValue` is a valid hex or it's a hex-like string the `Color.from(newValue)?.toHex6()` equal valid formatted hexString.

```ts
const hexValue = Color.from(newValue)?.toHex();
```

Reply:
Gotcha, updated to use toHex instead. Should the lower level utility be private?
Fallback added because from() returns possibly undefined